### ISSUE TITLE
`State.[cg]pu_local_snapshot.net_virial` invalid for particles `i >= 1`.

### ISSUE THREAD

**Issue by Oscar-Feng** [glotzerlab/hoomd-blue#1663]
### Description

I am facing an issue with the calculation of virials using the `snapshot.particles.net_virial` function from `gpu_local_snapshot` in HOOMD when simulating active Brownian particles with a Lennard-Jones potential. The virial outputs do not match the expected values for a minimal example of three particles.

I am simulating three particles with a Lennard-Jones potential using the following parameters:
- $\varepsilon$: 100
- $\sigma$ : $2r$
- Cutoff distance ($r_{\rm cut}$): $2^{1/6}\sigma$
- Particle diameter ($D$): $2r = 2$

The particles are initially placed at coordinates:
1. (0, 0, 0)
2. (0, 2.1, 0)
3. (2.1, 0, 0)

After a few simulation steps (to compute the particle data) so the particles have not moved so far, I expect the virials $(W^{xx}, W^{xy}, W^{xz}, W^{yy}, W^{yz}, W^{zz})$ of the three particles to be approximately as follows:
1. (A, 0, 0, B, 0, 0)
2. (0, 0, 0, B, 0, 0)
3. (A, 0, 0, 0, 0, 0)

Here, A and B are numbers on the order of $\sim 10^2$, and other values are relatively small.

### Script

```python
import os
import hoomd
import gsd.hoomd
import cupy

gpu = hoomd.device.GPU()
simulation = hoomd.Simulation(device=gpu,seed=12)

N = 3
radius=1

gsd_file = "3d_" + str(N) + "_.gsd"

frame = gsd.hoomd.Frame()
frame.particles.N = N
frame.particles.position = [(0,0,0),(0,2.1,0),(2.1,0,0)]
frame.particles.typeid = [0] * N
frame.particles.types = ['A']
frame.particles.mass = [1]*N
frame.particles.diameter = [2*radius] * N
frame.particles.moment_inertia=[(1,1,1)]*N
frame.configuration.box = [8, 8, 8, 0, 0, 0]

with gsd.hoomd.open(name=gsd_file, mode='w') as f:
        f.append(frame)

simulation.create_state_from_gsd(filename=gsd_file)

all = hoomd.filter.All()
time_step = 1e-4
simulation.always_compute_pressure = True

integrator = hoomd.md.Integrator(dt=time_step)
cell = hoomd.md.nlist.Cell(buffer=0.1)
lj = hoomd.md.pair.LJ(nlist=cell)
lj.params[('A', 'A')] = dict(epsilon=100, sigma= 2.0 * radius)
lj.r_cut[('A', 'A')] = 2.0 ** (1. / 6.) * 2 * radius
integrator.forces.append(lj)

langevin = hoomd.md.methods.Langevin(filter=hoomd.filter.All(),kT=1.0)
integrator.methods.append(langevin)
simulation.operations.integrator = integrator

simulation.run(15)

print(simulation.state.get_snapshot().particles.position)

with simulation.state.gpu_local_snapshot as snapshot:
            tot_virial= cupy.array(snapshot.particles.net_virial)
            tot_virial= tot_virial.get()
print(tot_virial)
```

### Output

```shell
[[-4.46463557e-04 -4.63498000e-04  3.48206777e-05]
 [-1.78023620e-05  2.10046411e+00  1.43657544e-05]
 [ 2.10047483e+00 -2.27464188e-05 -1.50416399e-05]]
[[1.8161403e-05 4.3628845e+02 4.3628842e+02 0.0000000e+00 0.0000000e+00 0.0000000e+00]
 [0.0000000e+00 0.0000000e+00 0.0000000e+00 0.0000000e+00 0.0000000e+00 0.0000000e+00]
 [0.0000000e+00 0.0000000e+00 0.0000000e+00 0.0000000e+00 8.9011550e-02 1.8054035e-01]]
```

### Expected output

[[1.8829017e+02 0.0000000e+00 0.0000000e+00 4.3625716e+02 0.0000000e+00 0.0000000e+00]
 [0.0000000e+00 0.0000000e+00 0.0000000e+00 4.3625716e+02 0.0000000e+00 0.0000000e+00]
 [1.8829017e+02 0.0000000e+00 0.0000000e+00 0.0000000e+00 0.0000000e+00 0.0000000e+00]]

where 0 can be replaced by relatively small numbers. I tried HOOMD v2 and its `net_virial()` did give me an output like this.

### Platform

GPU

### Installation method

Compiled from source

### HOOMD-blue version

4.3.0

### Python version

3.11.6

**Comment by joaander** [comment]
Langevin is not necessary to reproduce this behavior. Also `run(0)` is sufficient to compute the net virial.

Here is my modified test:
```
import hoomd
import gsd.hoomd
import numpy

cpu = hoomd.device.CPU()
simulation = hoomd.Simulation(device=cpu,seed=12)

N = 3
radius=1

gsd_file = "3d_" + str(N) + "_.gsd"

frame = gsd.hoomd.Frame()
frame.particles.N = N
frame.particles.position = [(0,0,0),(0,2.1,0),(2.1,0,0)]
frame.particles.typeid = [0] * N
frame.particles.types = ['A']
frame.particles.mass = [1]*N
frame.particles.diameter = [2*radius] * N
frame.particles.moment_inertia=[(1,1,1)]*N
frame.configuration.box = [8, 8, 8, 0, 0, 0]

with gsd.hoomd.open(name=gsd_file, mode='w') as f:
        f.append(frame)

simulation.create_state_from_gsd(filename=gsd_file)

simulation.always_compute_pressure = True

integrator = hoomd.md.Integrator(dt=0)
cell = hoomd.md.nlist.Cell(buffer=0.1)
lj = hoomd.md.pair.LJ(nlist=cell)
lj.params[('A', 'A')] = dict(epsilon=100, sigma= 2.0 * radius)
lj.r_cut[('A', 'A')] = 2.0 ** (1. / 6.) * 2
integrator.forces.append(lj)

simulation.operations.integrator = integrator

simulation.run(0)

with simulation.state.cpu_local_snapshot as snapshot:
    print("cpu_local_snapshot:")
    print(numpy.array(snapshot.particles.net_virial))
    print()

print("lj.virials:")
print(lj.virials)
print()

with lj.cpu_local_force_arrays as arrays:
    print("lj.cpu_local_force_arrays:")
    print(numpy.array(arrays.virial))
```

I get the output:
```
cpu_local_snapshot:
[[440.95181195   0.         440.95181195   0.           0.
    0.        ]
 [  0.           0.           0.           0.           0.
    0.        ]
 [  0.           0.           0.           0.           0.
    0.        ]]

lj.virials:
[[440.95181195   0.           0.         440.95181195   0.
    0.        ]
 [  0.           0.           0.         440.95181195   0.
    0.        ]
 [440.95181195   0.           0.           0.           0.
    0.        ]]

lj.cpu_local_force_arrays:
[[440.95181195   0.           0.         440.95181195   0.
    0.        ]
 [  0.           0.           0.         440.95181195   0.
    0.        ]
 [440.95181195   0.           0.           0.           0.
    0.        ]]
```

`Force.virials` and `Force.cpu_local_force_arrays` produce the expected output. `State.cpu_local_snapshot` and `State.gpu_local_snapshot` do not. @b-butler, please check `LocalParticleData` for errors compared to `LocalForceComputeData` and submit a fix.

**Comment by Oscar-Feng** [comment]
@joaander, thanks a lot for your help! I could use `Force.virials` and `Force.cpu_local_force_arrays` to generate the correct output. Looking forward to seeing the fix in a future version.

**Comment by joaander** [comment]
Fixed in #1663.
